test/TestCases/Expressions/RealImag.cpp

```diff
@@ -51,11 +51,28 @@ TEMPLATE_PRODUCT_TEST_CASE ("Imag", "[complex]", (PlatformVectorOps, VCTR_NATIVE
     const vctr::Vector imag = filter << vctr::imag << srcA;
     const vctr::Vector imagU = filter << vctr::imag << srcUnaligned;
 
-    // std::is_arithmetic is not true for complex values, so this checks if the return type is always imag valued, even for
+    // std::is_arithmetic is not true for complex values, so this checks if the return type is always real valued, even for
     // complex input
     static_assert (std::is_arithmetic_v<typename decltype (imag)::value_type>);
     static_assert (std::is_arithmetic_v<typename decltype (imagU)::value_type>);
 
     REQUIRE_THAT (imag, vctr::EqualsTransformedBy<getImagPart> (srcA));
     REQUIRE_THAT (imagU, vctr::EqualsTransformedBy<getImagPart> (srcUnaligned));
+}
+
+TEMPLATE_PRODUCT_TEST_CASE ("RealToComplex", "[complex]", (PlatformVectorOps, VCTR_NATIVE_SIMD), (float, double) )
+{
+    VCTR_TEST_DEFINES (10)
+
+    const vctr::Vector cplx = filter << vctr::realToComplex << srcA;
+    const vctr::Vector cplxU = filter << vctr::realToComplex << srcUnaligned;
```

Comment:
I assume this creates real complex vectors? If that is so, maybe let's also test with data with imag != 0?

Reply:
The new expression converts a source `Vector<float>` into a destination `Vector<std::complex<float>>` with all imaginary parts in the destination vector being zero. I'm not exactly sure what you mean by "test with data with imag != 0?" when the source is a real valued float vector?

Reply:
Sorry, I was momentarily confused and thought we are generating test data for a complex-to-real expression further below while this is already the expression under test. You can ignore my comment.